### ISSUE TITLE
ModuleNotFoundError: No module named 'packaging' regression in polars 0.16.6

### ISSUE THREAD

**Issue by jakob-keller** [pola-rs/polars#6966]
#6902 introduces a dependency on `packaging`, which is not part of the Python standard library and triggers a `ModuleNotFoundError` unless otherwise installed by chance.

https://github.com/pola-rs/polars/blob/fe613e1a5c4a033d2167272de39aa5ce12a417b4/py-polars/polars/internals/dataframe/frame.py#L107

This popped up for me when using polars within a docker container, not on a typical development machine. Let me know, if you need instructions to reproduce.

**Comment by ritchie46** [comment]
@alexander-beedie can we try to get that dependency out?

**Comment by alexander-beedie** [comment]
> @alexander-beedie can we try to get that dependency out?

Sure; up until about 30 minutes ago I didn't even realise it wasn't part of the standard distribution! I must have some implicit dependency/imports coming from elsewhere - having said that, so must we, or I'd have expected CI to fail...🤔 

Either way, will take care of it tonight/tomorrow.

**Comment by ritchie46** [comment]
Yes, I think we should add a CI job that tests imports without pyarrow, pandas etc installed.
